kmod/src/kernelcompat.h

```diff
@@ -100,4 +100,14 @@ static inline int dir_emit_dots(struct file *file, void *dirent,
 #define KC_RB_DECLARE_CALLBACKS RB_DECLARE_CALLBACKS
 #endif
 
+#ifdef KC_INODE_I_RWSEM
+#define kc_inode_lock(inode) down_write(inode->i_rwsem)
```

Comment:
After some more reading... `inode_lock()` and friends already existed and used `i_mutex` since like forever, so this becomes much more simple without any `#ifdef` use.

However, there is a name clash in `file.c` where there's a stack variable that needs to be renamed from `inode_lock` so I named it `scoutfs_inode_lock`. A little long but it at least it's simple.